Makefile

```diff
@@ -1,148 +1,26 @@
-	done
-	rm -rf out
-	cd ./dist && tar -czvf ../hook-${GIT_VERSION}.tar.gz ./*
-
-
-dev-dist: dev dev-convert
-	rm -rf ./dist ./convert
-	mkdir ./dist
-	mv ./initramfs-${ARCH}.gz ./dist/initramfs-${ARCH}
-	mv ./out/hook-${ARCH}-kernel ./dist/vmlinuz-${ARCH}
-	rm -rf out
-	cd ./dist && tar -czvf ../hook-${GIT_VERSION}.tar.gz ./*
-
-deploy: dist
-ifeq ($(shell git rev-parse --abbrev-ref HEAD),main)
-	s3cmd sync ./hook-${GIT_VERSION}.tar.gz s3://s.gianarb.it/hook/${GIT_VERSION}.tar.gz
-	s3cmd cp s3://s.gianarb.it/hook/hook-${GIT_VERSION}.tar.gz s3://s.gianarb.it/hook/hook-main.tar.gz
-endif
-
-.PHONY: clean
-clean:
-	rm ./hook-${GIT_VERSION}.tar.gz
-	rm -rf dist/ out/ tink-docker/local/ bootkit/local/
+help: ## Print this help
+	@grep --no-filename -E '^[[:alnum:]_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sed 's/:.*## /·/' | sort | column -t -W 2 -s '·' -c $$(tput cols)
 
-# BEGIN: lint-install ../hook
+-include rules.mk
+-include lint.mk
 
-GOLINT_VERSION ?= v1.42.0
-HADOLINT_VERSION ?= v2.6.1
-SHELLCHECK_VERSION ?= v0.7.2
-LINT_OS := $(shell uname)
-LINT_LOWER_OS  = $(shell echo $OS | tr '[:upper:]' '[:lower:]')
-LINT_ARCH := $(shell uname -m)
-GOLINT_CONFIG:=$(shell dirname $(realpath $(firstword $(MAKEFILE_LIST))))/.golangci.yml
+build: containers images ## Build boot files and container images for all supported architectures
 
-lint: out/linters/shellcheck-$(SHELLCHECK_VERSION)/shellcheck out/linters/hadolint-$(HADOLINT_VERSION) out/linters/golangci-lint-$(GOLINT_VERSION)
-	find . -name go.mod | xargs -n1 dirname | xargs -n1 -I{} sh -c "cd {} && golangci-lint run -c $(GOLINT_CONFIG)"
-	out/linters/shellcheck-$(SHELLCHECK_VERSION)/shellcheck $(shell find . -name "*.sh")
-	out/linters/hadolint-$(HADOLINT_VERSION) $(shell find . -name "*Dockerfile")
+dev: $(dev-deps) ## Build boot files and container images for currently running architecture
 
-out/linters/shellcheck-$(SHELLCHECK_VERSION)/shellcheck:
-	mkdir -p out/linters
-	curl -sfL https://github.com/koalaman/shellcheck/releases/download/v0.7.2/shellcheck-$(SHELLCHECK_VERSION).$(LINT_OS).$(LINT_ARCH).tar.xz | tar -C out/linters -xJf -
+images: $(image-deps) ## Build boot files for all supported architectures
 
-out/linters/hadolint-$(HADOLINT_VERSION):
-	mkdir -p out/linters
-	curl -sfL https://github.com/hadolint/hadolint/releases/download/v2.6.1/hadolint-$(LINT_OS)-$(LINT_ARCH) > out/linters/hadolint-$(HADOLINT_VERSION)
-	chmod u+x out/linters/hadolint-$(HADOLINT_VERSION)
+containers: build-hook-bootkit build-hook-docker ## Build container images
 
-out/linters/golangci-lint-$(GOLINT_VERSION):
-	mkdir -p out/linters
-	curl -sfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b out/linters $(GOLINT_VERSION)
-	mv out/linters/golangci-lint out/linters/golangci-lint-$(GOLINT_VERSION)
+debug-images: $(debug-image-deps) ## Build ssh enabled boot files for currently running architecture
 
-# END: lint-install ../hook
+run: run-$(ARCH) ## Boot system using qemu 
```

Comment:
This fails for me with `error: failed to solve: rpc error: code = Unknown desc = authorization status: 401: authorization failed` so there's some sort of implicit assumption about pushing that is broken when I try to test...

Reply:
:thinking: I'll see if I can track this down.

Reply:
argh, yeah I'm able to track this down. Its mostly having to do with how prod/dev/debug builds happen. run depends on the target that pushes to $registry. That doesn't really make any sense to me but I didn't want to change that behavior. I caused the push to fail though which imo is correct behavior since we don't have versioned tags and thus the pushes end up overwriting "released" images. I'd like to sort of leave it like this for now and come back with a follow up changing to versioned images and related stuff.